src/main/java/net/datafaker/transformations/JsonTransformer.java

```diff
@@ -53,31 +53,43 @@ public String generate(Iterable<IN> input, Schema<IN, ?> schema) {
             throw new IllegalArgumentException("The sequence should be finite of size");
         }
 
-        StringJoiner data = new StringJoiner(LINE_SEPARATOR);
+        StringBuilder sb = new StringBuilder();
         Iterator<IN> iterator = input.iterator();
-        while (iterator.hasNext()){
-            data.add(apply(iterator.next(), schema) + (commaBetweenObjects && iterator.hasNext() ? "," : ""));
+        int size = 0;
+        while (iterator.hasNext()) {
+            size++;
+            sb.append(apply(iterator.next(), schema)).append(commaBetweenObjects && iterator.hasNext() ? "," : "");
         }
 
-        return data.length() > 1 ? wrappers[0] + LINE_SEPARATOR + data + LINE_SEPARATOR + wrappers[1]
-            : data.toString();
+        String result = size > 1 ? "" + WRAPPERS[0] + sb + WRAPPERS[1] : sb.toString();
+
+        return prettPrint ? result.startsWith("{")
+            ? new JSONObject(result).toString(2) : new JSONArray(result).toString(2) : result;
     }
 
     @Override
     public String generate(Schema<IN, ?> schema, int limit) {
         StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < limit; i++) {
-            sb.append(apply(null, schema, i));
-            if (commaBetweenObjects && i < limit - 1) {
-                sb.append(",").append(LINE_SEPARATOR);
+
+        if (limit == 1) {
+            sb.append(apply(null, schema, 0));
+            return sb.toString();
+        } else {
+            for (int i = 0; i < limit; i++) {
+                sb.append(apply(null, schema, i));
+                if (commaBetweenObjects && i < limit - 1) sb.append(",");
             }
         }
-        return limit > 1 ? wrappers[0] + LINE_SEPARATOR + sb + LINE_SEPARATOR + wrappers[1] : sb.toString();
+
+        String result = "" + WRAPPERS[0] + sb + WRAPPERS[1];
+
+        return prettPrint ? result.startsWith("{")
+            ? new JSONObject(result).toString(2) : new JSONArray(result).toString(2) : result;
```

Comment:
This is a breaking change 
and this is not clear why we need it.

The main idea behind was that the output was in JSON Lines format (https://jsonlines.org/)

And the main feature is that it could be easily copied and pasted into BigQuery as stated in their doc (https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-json)

Now i don't see how it could be possible

Reply:
I understand your concern @snuyanzin, and I do agree it's a breaking change as the way to generate large datasets is now missing.

First, the "regular" JSON output was wrong as it wasn't taking the correct format, based on its semantics. See [JSON](https://www.json.org/json-en.html).
The output was, all the time, an object even when the output was an array, and this change fixed it.

Now, from my point of view, the way this feature was created this biased in only one usage, which might be not the common usage for the Datafaker consumers. Datafaker has delivered this feature thinking about generating JSON output to be processed for large external datasets or tools, like logging systems and the mentioned BigQuery example. And it's awesome!

As an end user, my first way of thinking is that Datafaker will generate a valid JSON for most of the use cases.
Nonetheless, it wasn't creating a valid JSON for any other purpose. The https://github.com/datafaker-net/datafaker/issues/1109 puts some light on it.

I think that a decision should be made, in terms of which format Datafaker will support it.
It can be one, it can be both.

Maybe a new builder in the `JsonTransformerBuilder` can be added to take care of the large dataset output.

I will follow any decision you guys make, but with the note that the current approach seems not valid for most Datafaker users.

Reply:
May be i was not clear enough, sorry
My point not about valid not valid json

Fixing json is ok

The problem is that here we change output format: now there either everything in one line or pretty

at the same time `jsonline` assumes that there should be one top level entity per line and I don't know how can we have with the proposed solution. 

So do we really need this `json ` lib usage? I guess we could fix wrong json issue without breaking existing approach

Reply:
Sorry for my misunderstanding.

The library is not that important if we make sure the JSON is correctly created. I intended to have this as a double-check, plus the pretty output.

I would say that we can keep all, making the pretty not default. By the way, when we apply the approach without commas, which is the JSON line solution, the pretty is not applied, but I have changed the code to have it in one line.

Can I propose to remove the one-line approach and keep the pretty output?

Reply:
Ideally, I would like the library to be removed. If that means no pretty printing, then I don't mind, I think that's a nice to have, and I don't think that's the responsibility of Datafaker.

Reply:
The problem with pretty print: measurement show that it 3 times slow for 1, 10, 1000 entities
```
Benchmark                              Mode  Cnt     Score    Error   Units
DatafakerSimpleMethods.json1          thrpt   10  2517.912 ± 43.562  ops/ms
DatafakerSimpleMethods.json10         thrpt   10   252.746 ±  3.304  ops/ms
DatafakerSimpleMethods.json100        thrpt   10    25.772 ±  0.437  ops/ms
DatafakerSimpleMethods.json100Pretty  thrpt   10     8.306 ±  0.013  ops/ms
DatafakerSimpleMethods.json10Pretty   thrpt   10    77.777 ±  2.426  ops/ms
DatafakerSimpleMethods.json1Pretty    thrpt   10  2513.785 ± 14.620  ops/ms
```

Reply:
No problem.
I will remove the library and other changes, keeping the original fix.